### ISSUE TITLE
Multiple image search

### ISSUE THREAD

**Issue by jpeltonen** [rainmanwy/robotframework-SikuliLibrary#25]
Would be nice to have keyword that would search for multiple images from screen. For example functionality of 'Wait Until Screen Contain' and 'Wait Until Screen Not Contain' in single keyword. Sometimes you need wait for 'success' image and 'failure' image on screen at the same time. If the 'failure' image could be detected in addition to 'success' image then you shouldn't have to wait for timeout in case of just being able to use 'Wait Until Screen Contain    success.png'.

Keyword could be something like 'Wait For Multiple Images    listOfWantedImagesToWait    listOfNotWantedImagesToWait'

**Comment by rainmanwy** [comment]
I am not sure how to use observe to wait two images. have you tried this with python?

**Comment by jpeltonen** [comment]
I haven't tried yet. I think it should also work in Java and would be great feature.

**Comment by rainmanwy** [comment]
ok, thanks! I will check this later. So keywork like "'Wait For Images" is enough, right?

**Comment by jpeltonen** [comment]
Yes it would be good to start with.

**Comment by rainmanwy** [comment]
One question is about the return value of this keyword.

Wait For Images  |  wantedImageToWait |  notWantedImageToWait

So the return value is not needed? just throw exception, for example:

if(find(notWantedImageToWait) || timeoutHappened(wantedImageToWait))
    throw exception

am i correct?

**Comment by jpeltonen** [comment]
Yes.

**Comment by rainmanwy** [comment]
i found this faq https://answers.launchpad.net/sikuli/+faq/1568, i may follow this faq to implement the keyword recently.

**Comment by rainmanwy** [comment]
@jpeltonen , i have add a keyword

    @RobotKeyword("Wait For Image"
            + "\n\n Check wantedImage exist. If notWantedImage appear or timeout happened, throw exception"
            + "\n @wantedImage: expected image in screen"
            + "\n @notWantedImage: unexpected image in screen"
            + "\n @timeout: wait sencods"
            + "\n\n Example Usage:"
            + "\n | Wait For Image  | wanted.png | notWanted.png | 5 |")

you may have a try, and give your comments. Thank you very much!

**Comment by jpeltonen** [comment]
Could you change like this: public void waitForImage(String wantedImage, String notWantedImage, int timeout) -> public void waitForImage(List<String> wantedImages, List<String> notWantedImages, int timeout)

**Comment by rainmanwy** [comment]
Hi, @jpeltonen . For multi images check, i am not sure the check points. For example, 
1. one of wantedImages exist, then keyword success, 
2. or all of wantedimages exist, then keyword success
...
i guess different users may have different understanding or expectation about this. 

So my proposal is provide a keyword to expose "exists" method to robot framework. So that you could implement your own keywords as robot user keywords or python keywords. How do you think?

**Comment by jpeltonen** [comment]
Yes maybe better idea to use "exist".

**Comment by jpeltonen** [comment]
Replaced 'Wait For Image' with 'Wait For Multiple Images' keyword in pull request [https://github.com/rainmanwy/robotframework-SikuliLibrary/pull/49/commits/80cc6232d4756a6f48a1955b9ffaace90a5ba2d9](url)

If expected images exist, then keywords succeeds. If one of not excpected images exists then keyword fails. 

Not expected images start index is used for selecting expected and not expected images in single image list passed to keyword. Robot Framework showed error with multiple list arguments in keyword so had to use shared list.

**Comment by jpeltonen** [comment]
Could also add optional parameter for selection 2. What do you think?

1. one of wantedImages exist, then keyword success,
2. or all of wantedimages exist, then keyword success

**Comment by rainmanwy** [comment]
Yes. I think this option could make the functionality of this keyword more clear. And i think the doc should be updated to describe how this keyword work.
________________________________
发件人: Jarkko Peltonen <notifications@github.com>
发送时间: 2017年10月25日 3:39
收件人: rainmanwy/robotframework-SikuliLibrary
抄送: Rainmanwy; Comment
主题: Re: [rainmanwy/robotframework-SikuliLibrary] Multiple image search (#25)


Could also add optional parameter for selection 2. What do you think?

  1.  one of wantedImages exist, then keyword success,
  2.  or all of wantedimages exist, then keyword success

―
